server-spi/src/main/java/org/keycloak/rar/AuthorizationDetailsJSONRepresentation.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright 2022 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.keycloak.rar;
+
+import com.fasterxml.jackson.annotation.JsonAnyGetter;
+import com.fasterxml.jackson.annotation.JsonAnySetter;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * The JSON representation of a Rich Authorization Request's "authorization_details" object.
+ *
+ * @see {@link <a href="https://datatracker.ietf.org/doc/html/draft-ietf-oauth-rar#section-2">Request parameter "authorization_details"</a>}
+ *
+ * @author <a href="mailto:dgozalob@redhat.com">Daniel Gozalo</a>
+ */
+public class AuthorizationDetailsJSONRepresentation implements Serializable {
```

Comment:
Minor: Maybe we can move this class to the "core" package together with classes like for example `AccessToken` ? There is a chance that this class might be used by applications/adapters and hence it may fit better into that place IMO.

Reply:
Good point. I wasn't sure about the best place to put some of these classes. As you said, this may be used by applications and adapters to create the RAR representation to be sent to the server, so I'm moving it.